Registry container is failing again on startup via docker-compose

Currently facing following error during registry startup in docker-compose:

```
registry_1  | Traceback (most recent call last):
registry_1  |   File "/usr/lib64/python3.7/runpy.py", line 193, in _run_module_as_main
registry_1  |     "__main__", mod_spec)
registry_1  |   File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
registry_1  |     exec(code, run_globals)
registry_1  |   File "/app/karapace/schema_registry_apis.py", line 2, in <module>
registry_1  |     from karapace.compatibility import Compatibility, IncompatibleSchema
registry_1  |   File "/app/karapace/compatibility.py", line 7, in <module>
registry_1  |     from karapace.schema_reader import SchemaType, TypedSchema
registry_1  |   File "/app/karapace/schema_reader.py", line 15, in <module>
registry_1  |     from karapace.utils import json_encode
registry_1  |   File "/app/karapace/utils.py", line 13, in <module>
registry_1  |     import karapace.rapu
registry_1  |   File "/app/karapace/rapu.py", line 10, in <module>
registry_1  |     from karapace.utils import json_encode
registry_1  | ImportError: cannot import name 'json_encode' from 'karapace.utils' (/app/karapace/utils.py)
```

Only Kafka and Zookeeper starts and operates fine.

I used codebase from 'master' branch.

Any thoughts on this?

So my steps are:

1. `cp container/start.sh start.sh`
2. `cp container/Dockerfile Dockerfile`
3. change the following line in Dockerfile from `ADD container/start.sh /opt` to `ADD start.sh /opt`
4. `docker build -t karaspace .`
5. `docker run -p 8081:8081 -p 8082:8082 karaspace:latest all` throws the following exception

[2020-09-11 10:27:06,825] INFO [KafkaServer id=0] started (kafka.server.KafkaServer)
Traceback (most recent call last):
  File "/usr/lib64/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/app/karapace/schema_registry_apis.py", line 2, in <module>
    from karapace.compatibility import Compatibility, IncompatibleSchema
  File "/app/karapace/compatibility.py", line 7, in <module>
    from karapace.schema_reader import SchemaType, TypedSchema
  File "/app/karapace/schema_reader.py", line 17, in <module>
    from karapace.utils import json_encode
  File "/app/karapace/utils.py", line 13, in <module>
    import karapace.rapu
  File "/app/karapace/rapu.py", line 10, in <module>
    from karapace.utils import json_encode


If you instead go back to the 2.0.1 release (3dc8d7bac61bb433bd37c9f1eac74d305d138b63)
1. git checkout -b 2.0.1 3dc8d7bac61bb433bd37c9f1eac74d305d138b63
2. `docker build -t karaspace .`
3. `docker run -p 8081:8081 -p 8082:8082 karaspace:latest all` 

This works

However if you pull the latest 2.0.1 release artifact and add the Dockerfile and start.sh and build `docker build -t karaspace:2.0.1 .` it fails to build

Step 15/16 : RUN make
 ---> Running in c6280c05ca0e
python3 version.py karapace/version.py
version.py:7: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "version.py", line 58, in <module>
    get_project_version(sys.argv[1])
  File "version.py", line 46, in get_project_version
    short_ver = [line.split("=", 1)[1].strip() for line in lines if line.startswith("short_ver")][0]
IndexError: list index out of range
make: *** [Makefile:38: karapace/version.py] Error 1

--- a/karapace/utils.py
+++ b/karapace/utils.py
@@ -10,7 +10,6 @@ from urllib.parse import urljoin
 import datetime
 import decimal
 import json as jsonlib
-import karapace.rapu
 import logging
 import requests
 import types
@@ -186,4 +185,5 @@ def convert_to_int(object_: dict, key: str, content_type: str):
     try:
         object_[key] = int(object_[key])
     except ValueError:
-        karapace.rapu.http_error(f"{key} is not a valid int: {object_[key]}", content_type, 500)
+        from karapace.rapu import http_error
+        http_error(f"{key} is not a valid int: {object_[key]}", content_type, 500)

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -7,3 +7,5 @@ requests==2.23.0
 pre-commit>=2.2.0
 aiokafka==0.6.0
 jsonschema==3.2.0
+pycodestyle==2.6.0
+isort==4.3.10

